Nonlinear_Model_Predictive_for_WECCCOMP/controller_init.m

```diff
@@ -0,0 +1,132 @@
+% init file for the WecSim model of the Wavestart arm
+close all; clc;
+% Clear persistent variables defined in: extractedPower, NMPC, observer_kf, and predictor.
+clear extractedPower NMPC observer_kf predictor;  
+
+%% %%% Model Parameters to build the Continous Matrices Ac, Bc and Cc for the Anaylisis of the WEC System
+a_CAB_neutral   = 1.170165;               % Angle CAB at equilibrium position
+L_AB            = 0.412;                  % Distance AB [m]
+L_AC            = 0.2;                    % Distance AC [m]
+L_BC_neutral    = 0.381408;               % Distance CB at equilibrium position [m]
+LAfloat         = 0.54875;                % Distance from pivot point A to the Cg of the float [m]
+J               = 1.04;                   % Inertia of arm and float [kg m^2]
+Jinf            = 0.4805;                 % Added inertia [kg m^2]
+Jt              = J + Jinf;               % Total inertia [kg m^2]
+Khs             = 92.33;                  % Hydrostatic stiffness coefficient [Nm rad^-1]
+bv              = 1.8;                    % Linear damping coefficient [N m (rad/s)^-1]
+if ~exist("dt","var")
+    dt    = simu.dt;                      % Sampling time [s]
+end
+
+%% %%%%%% State Space Matrices for the Linear system %%%%%%%%%%%%%%%%%%%%%%%
+load('./waveData/radM_matrices.mat')      % Load state-space model matrices for the radiation moment around point A.
+nx_radM = size(Ar,2);
+Ac      = [0,                1,             zeros(1,nx_radM);
+           -Khs./Jt,          -(Dr+bv)./Jt,  -Cr./Jt;
+           zeros(nx_radM,1),  Br,            Ar];      
+nx      = size(Ac,1);
+Bc      = [ 0; 1/Jt; zeros(nx_radM,1) ];                                                                                                                      
+Cc      = [1,zeros(1, nx - 1);
+           0,1,zeros(1, nx - 2)];
+Dc      = zeros(size(Cc,1),size(Bc,2));
+sysC    = ss(Ac,Bc,Cc,Dc);                % Declares Continuous State Space

```

Comment:
@Liam-Guerrero It may be helpful that each time a function that uses additional MATLAB/Simulink dependencies to comment in the code so users know this is an extra function outside of the base WEC-Sim code.